src/Framework/MSBuildEventSource.cs

```diff
@@ -361,6 +370,32 @@ public void MSBuildExeStop(string commandLine)
             WriteEvent(46, commandLine);
         }
 
+        [Event(47)]
+        public void ExecuteTaskStart(string taskName)
+        {
+            WriteEvent(47, taskName);
+        }
+
+        [Event(48)]
+        public void ExecuteTaskStop(string taskName, string nanosecondsElapsed)
+        {
+            WriteEvent(48, taskName, nanosecondsElapsed);
+        }
+
+        #endregion
+
+        #region Non-Events
+        [NonEvent]
+        public static long GetThreadTime()
+        {
+            long kernelTime, userTime;
+            if (!Convert.ToBoolean(GetThreadTimes(GetCurrentThread(), out _, out _, out kernelTime, out userTime)))
```

Comment:
Is it a realistic scenario that the values for kernelTime and userTime would overflow? If so, how can we test for that?

Reply:
See above comment. I think the fact that Duration() made them all non-negative in building MSBuild.sln means there was overflow, and it is now corrected.

Reply:
I'd recommend staying out of the business of doing this type of calculation and just use the trace time stamps to calculate the time.  Also note that this won't work on Unix as it depends on Windows-specific pinvokes.